src/shapes.ts

```diff
@@ -25,6 +25,10 @@ export interface ProviderIdentity extends Identity {
     runtimeUuid?: string;
 }
 
+export interface EntityIdentity extends Identity {
```

Comment:
It just adds an extra property 'isExternal'.  Can we make this property as an optional one in Identity so we don't need this new type?

Reply:
It's probably better to only have this property defined where it might be set rather than changing identity everywhere. Makes the type more meaningful.

Reply:
Agree to make the type more meaningful.